src/hotspot/share/opto/node.hpp

```diff
@@ -1128,6 +1128,23 @@ class Node {
   // Set control or add control as precedence edge
   void ensure_control_or_add_prec(Node* c);
 
+  // Visit boundary uses of the node and apply a callback function for each.
+  // Recursively traverse uses, stopping and applying the callback when
+  // reaching a boundary node, defined by is_boundary.
+  // Note: the function definition appears after the complete type definition
+  // of Unique_Node_List.
+  template <typename Callback, typename Check>
+  void visit_uses(Callback callback, Check is_boundary);
+
+  // Visit all non-cast uses of the node, bypassing ConstraintCasts.
+  // Pattern: this (-> ConstraintCast)* -> non_cast
+  // In other words: find all non_cast nodes such that
+  // non_cast->uncast() == this.
```

Comment:
same here

Reply:
This is the original comment for `visit_uncasted_uses` (I haven't changed it). But, I'm fine with changing it to something better (see my question above as well).